File: Level2/221102/Java/RemovePair.java
import java.util.Stack;

public class RemovePair {
  public int solution(String s) {
    Stack<Character> stack = new Stack<>();

    for (char word : s.toCharArray()) {
      if(stack.isEmpty() || stack.peek() != word) {
        stack.add(word);
      }
      else {
        stack.pop();
      }
    }

    int answer = 0;

    if (stack.isEmpty()) {
      answer = 1;
    }
    return answer;
  }
}

File: Programmers/Level2/221103/JavaScript/RemovePair.js
export function toArray(s) {
  return s.split('');
}

export function remove(words) {
  const stack = [];

  words.map((word) => {
    if (stack.length === 0 || stack[stack.length - 1] !== word) {
      stack.push(word);
      return;
    }
    if (stack[stack.length - 1] === word) {
      stack.pop();
    }
  });

  return stack;
}

export function solution(s) {
  const words = toArray(s);

  const stackArray = remove(words);

  return stackArray.length === 0 ? 1 : 0;
}
